generate-release/src/release_notes.rs

```diff
@@ -107,3 +127,81 @@ file_name = "{file_name}.md"
         title = title.trim().replace('"', "\\\"")
     )
 }
+
+/// This function:
+///
+/// 1. Generates a new issue on the `bevy-website` repo for the given PR.
+/// 2. Leaves a comment in the original PR linking to the new issue.
+///
+/// If the issue already exists, no action is taken.
+#[allow(clippy::too_many_arguments)]
+fn generate_and_open_issue(
+    client: &GithubClient,
+    existing_issue_titles: &HashSet<String>,
+    pr: &GithubIssuesResponse,
+    title: &str,
+    file_path: &Path,
+    dry_run: bool,
+) {
+    let pr_number = pr.number;
+    let issue_title = format!("Write release notes for PR #{pr_number}: {title}");
+
+    // Check if this issue already exists
+    // If it does, we don't want to spam the repo with duplicate issues
+    if existing_issue_titles.contains(&issue_title) {
+        println!("Issue already exists for PR #{}: {}", pr_number, title);
+        return;
+    }
+
+    let pr_url = format!("https://github.com/bevyengine/bevy/pull/{pr_number}",);
+    let file_path = file_path.to_string_lossy();
+
+    // The weird indentation is intentional.
+    // Otherwise, the text is not formatted correctly on GitHub as the tabs are copied into the issue body.
+    let issue_body = format!(
+        "{pr_url} needs release notes for the upcoming Bevy release!
+
+Please reply below if you'd like to write these notes. 
+While the author(s) of the PR often have the context, knowledge and motivation to draft the release notes for their feature, anyone can contribute release notes!
+
+------
+
+Release notes should:
+
+1. Clearly motivate the change.
+2. Be written in a way that is understandable by the average Bevy user: some programming background and a general understanding of games.
+3. Show off the coolest features of the PR. Screenshots are awesome, but elegant APIs are also welcome!
+4. If this was a perf-centric PR, quantify the performance improvements. Graphs and statistics work well for this.
+
+We can help you revise the release notes: a rough draft alone is incredibly useful :)
+Your expertise is invaluable for contextualizing the changes; we'll work with you to bring the technical writing up to par.
+
+To submit your release notes, modify `{file_path}` and submit a PR.
+In that PR, please mention this issue with the `Fixes #ISSUE_NUMBER` keyphrase so it gets closed automatically.");
+
+    let labels = vec!["A-Release-Notes", "C-Content", "S-Ready-For-Implementation"];
+
+    if dry_run {
+        println!("Would open issue on GitHub:");
+        println!("Title: {}", issue_title);
+        println!("Body: {}", issue_body);
+        println!("Labels: {:?}", labels);
```

Comment:
Nit: Could we include the comment output here as well?

Reply:
Not really unfortunately. The critical information (which issue to link to) isn't available until we actually create the issue and get the response, which we can then parse for the link.

Reply:
Makes sense! It can always be added in a future PR (i.e. printing out the comment with a dummy link or something).